client/data/help/use-help-search-query.ts

```diff
@@ -1,26 +1,90 @@
-import type { LinksForSection } from '@automattic/data-stores';
+import { useQueryClient, useQuery } from '@tanstack/react-query';
+import apiFetch from '@wordpress/api-fetch';
+import wpcomRequest, { canAccessWpcomApis } from 'wpcom-proxy-request';
+
+export interface SearchResult {
+	link: string;
+	title: string | React.ReactChild;
+	content?: string;
+	icon?: string;
+	post_id?: number;
+	blog_id?: number;
+	source?: string;
+}
+
+interface APIFetchOptions {
+	global: boolean;
+	path: string;
+}
+
 export const useHelpSearchQuery = (
 	search: string,
-	queryOptions: Record< string, unknown > = {}
+	locale = 'en',
+	queryOptions: Record< string, unknown > = {},
+	sectionName = ''
 ) => {
-	const locale = useLocale();
-	const params = new URLSearchParams( {
-		include_post_id: '1',
-		locale,
-	} );
+	const queryClient = useQueryClient();
 
-	if ( search ) {
-		params.append( 'query', search );
+	// set up article list fetching
+	let fetchArticlesAPI: () => Promise< unknown >;
+	const params = `?query=${ encodeURIComponent( search ) }&locale=${ encodeURIComponent(
+		locale
+	) }&section=${ encodeURIComponent( sectionName ) }`;
+	if ( canAccessWpcomApis() ) {
+		fetchArticlesAPI = () =>
+			wpcomRequest( {
+				path: `help/search/wpcom${ params }`,
+				apiNamespace: 'wpcom/v2/',
+				apiVersion: '2',
+			} );
+	} else {
+		fetchArticlesAPI = () =>
+			apiFetch( {
+				global: true,
+				path: `/help-center/search${ params }`,
+			} as APIFetchOptions );
 	}
 
-	return useQuery< { wordpress_support_links: LinksForSection[] } >( {
-		queryKey: [ 'help', search ],
-		queryFn: () =>
-			wpcomRequest( { path: '/help/search', query: params.toString(), apiVersion: '1.1' } ),
-		enabled: !! search,
+	// set up article fetching
+	let fetchArticleAPI: ( blogId: number, postId: number ) => Promise< { content: string } >;
+	if ( canAccessWpcomApis() ) {
+		fetchArticleAPI = ( blogId: number, postId: number ) =>
+			wpcomRequest( {
+				path: `help/article/${ blogId }/${ postId }`,
+				apiNamespace: 'wpcom/v2/',
+				apiVersion: '2',
+			} );
+	} else {
+		fetchArticleAPI = ( blogId: number, postId: number ) =>
+			apiFetch( {
+				global: true,
+				path: `/help-center/fetch-post?blog_id=${ blogId }&post_id=${ postId }`,
+			} as APIFetchOptions );
+	}
+
+	return useQuery< any >( {
+		queryKey: [ 'help-center-search', search, sectionName ],
+		queryFn: fetchArticlesAPI,
+		onSuccess: async ( data ) => {
+			const newData = await Promise.all(
+				data.map( async ( result: SearchResult ) => {
+					// add the content if we don't have it, and have enough info to fetch it
+					if ( ! result.content && result.blog_id && result.post_id ) {
+						const article: { [ content: string ]: string } = await fetchArticleAPI(
+							result.blog_id,
+							result.post_id
+						);
+						return { ...result, content: article.content };
+					}
+					return result;
+				} )
+			);
+
+			queryClient.setQueryData( [ 'help', search ], newData );
```

Comment:
Does this query key exist? I can't find it. If you're trying to alter the results of the query itself (and not another query), [select](https://tanstack.com/query/v4/docs/react/reference/useQuery#:~:text=select%3A%20(data%3A%20TData)%20%3D%3E%20unknown) is better, IMO.

Reply:
@alshakero Not sure I understand (this isn't something I've changed here - it's already in [the existing hook](https://github.com/Automattic/wp-calypso/blob/trunk/packages/help-center/src/hooks/use-help-search-query.ts))

Reply:
What this line does, it sets the data of a query that has the same key. Eg

```js
useQuery( [ 'help', search ], ... );
```

And I searched Calypso and it doesn't exist, so I imagine it's a leftover. It's probably not doing anything here.